### PR TITLE
Skeleton for WPT tests

### PR DESCRIPTION
**Pull request by lemire**
This adds a WPT test executed, hooked into CMake.

For now, it just loads the JSON files and parses their content, printing the results to the string.

Once the parser is actually implemented, it should be possible to just fill-in the blanks.

Note that two of the JSON files are not valid JSON as per the RFC since they are not valid UTF-8. When that happens, we simply grab a raw string (effectively just pointing inside the input document). It is somewhat odd: for percent encoding, the specification appears to [require that the input be UTF-8](https://url.spec.whatwg.org/#url-parsing). It is probably worth making sure that these test cases are usable and follow the specification. In any case, the skeleton still provides the data.

It also adds CI.

Fixes https://github.com/ada-url/ada/issues/3

### CODE REVIEW

File: .github/workflows/ubuntu.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+name: Ubuntu 22.04 CI (GCC 12)
+
+on:
+  push:
+    branches:
+      - master
```
Comment (anonrig) [inline .github/workflows/ubuntu.yml:6]: This should be main

File: tests/CMakeLists.txt [hunk 2]
```diff
@@ -1,7 +1,27 @@
+
+include(FetchContent)
+FetchContent_Declare(
+  simdjson
+  GIT_REPOSITORY https://github.com/simdjson/simdjson.git
```
Comment (anonrig) [inline tests/CMakeLists.txt:5]: Do we need “tools/update-wpt.sh” after this change?

File: .github/workflows/ubuntu.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+          - {shared: OFF}
+    steps:
+      - uses: actions/checkout@v3
+      - name: Use cmake
+        run: |
+          mkdir build &&
+          cd build &&
+          CXX=g++-12 cmake -DBUILD_SHARED_LIBS=${{matrix.shared}} ..  &&
+          cmake --build .   &&
+          ctest -j --output-on-failure
```
Comment (anonrig) [inline .github/workflows/ubuntu.yml:30]: Newline is missing from this document

File: tests/wpt_tests.cpp [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+      }
+    }
+  }
+  TEST_SUCCEED()
+}
+
+int main() {
+  std::cout << "Running WPT tests.\n" << std::endl;
+
+  if (percent_encoding() && setters_tests_encoding() && toascii_encoding() &&
```
Comment (anonrig) [inline tests/wpt_tests.cpp:243]: I might be wrong here but: I believe this does not run `setters_tests_encoding()` if `percent_encoding()` fails. Can we make sure each test runs regardless?

File: .github/workflows/ubuntu.yml (hunk 1 above)
Comment (lemire) [reply .github/workflows/ubuntu.yml:6]: good catch!

File: .github/workflows/ubuntu.yml (hunk 3 above)
Comment (lemire) [reply .github/workflows/ubuntu.yml]: Right. Fixed.

File: tests/wpt_tests.cpp (hunk 4 above)
Comment (lemire) [reply tests/wpt_tests.cpp]: If you'd like. Changed by the next commit.

**Comment by lemire** [comment]
@anonrig I have added in CI pedantic tests that will fail (by design) currently since there are compiler warnings. I will not fix them as part of this CI (otherwise, it would be too invasive).

**Pull request merged**